includes/settings/class-amp-settings-post-types.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Post types settings.
+ *
+ * @package AMP
+ */
+
+/**
+ * Settings post types class.
+ */
+class AMP_Settings_Post_Types {
+
+	/**
+	 * Settings section id.
+	 *
+	 * @var string
+	 */
+	protected $section_id = 'post_types';
+
+	/**
+	 * Settings config.
+	 *
+	 * @var array
+	 */
+	protected $setting = array();
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->setting = array(
+			'id'          => 'post_types_support',
+			'label'       => __( 'Post Types Support', 'amp' ),
+			'description' => __( 'Enable/disable AMP post type(s) support', 'amp' ),
+		);
+	}
+
+	/**
+	 * Initialize.
+	 */
+	public function init() {
+		add_action( 'admin_init', array( $this, 'register_settings' ) );
+		add_action( 'update_option_' . AMP_Settings::SETTINGS_KEY, 'flush_rewrite_rules' );
+	}
+
+	/**
+	 * Register the current page settings.
+	 */
+	public function register_settings() {
+		register_setting( AMP_Settings::SETTINGS_KEY, AMP_Settings::SETTINGS_KEY );
```

Comment:
This should add a `sanitize_callback` here for good measure, even though it is only ever read out with a cast to boolean.